pkg/buildah/merge.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2022 cuisongliu@qq.com.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildah
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/containers/buildah"
+	buildahcli "github.com/containers/buildah/pkg/cli"
+	"github.com/containers/buildah/util"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/labring/sealos/pkg/buildimage"
+
+	"github.com/labring/sealos/pkg/utils/logger"
+	"github.com/labring/sealos/pkg/utils/rand"
+)
+
+func newMergeCommand() *cobra.Command {
+	layerFlagsResults := buildahcli.LayerResults{}
+	buildFlagResults := buildahcli.BudResults{}
+	fromAndBudResults := buildahcli.FromAndBudResults{}
+	userNSResults := buildahcli.UserNSResults{}
+	namespaceResults := buildahcli.NameSpaceResults{}
+	buildahInfo := &buildah.BuilderInfo{}
+	sopts := saveOptions{}
+	mergeCommand := &cobra.Command{
+		Use:   "merge",
+		Short: "merge multiple images into one",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			br := buildahcli.BuildOptions{
+				LayerResults:      &layerFlagsResults,
+				BudResults:        &buildFlagResults,
+				UserNSResults:     &userNSResults,
+				FromAndBudResults: &fromAndBudResults,
+				NameSpaceResults:  &namespaceResults,
+			}
+			logger.Debug("save enable: %+v", sopts.enabled)
+			return buildCmd(cmd, []string{buildahInfo.MountPoint}, sopts, br)
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			arch := getArchFromFlag(cmd)
+			oss := getOSFromFlag(cmd)
+			variant := getVariantFromFlags(cmd)
+			tag := getTagFromFlags(cmd)
+			logger.Debug("os: %s, arch: %s, variant: %s, tag: %+v", oss, arch, variant, tag)
+
+			platform := v1.Platform{
+				Architecture: arch,
+				OS:           oss,
+				Variant:      variant,
+			}
+			var err error
+			buildahInfo, err = mergeImagesWithScratchContainer(tag[0], args, platform)
+			if err != nil {
+				return err
+			}
+			if err := setDefaultFlagsWithSetters(cmd, setDefaultSaveImageFalseFlag); err != nil {
```

Comment:
no necessary to define a function since is not a very common use. just `return setDefaultFlagsWithSetters(cmd, setDefaultFlagIfNotChanged(c, "save-image", "false"))`.

Reply:
done